app/controllers/omniauth_controller.rb

```diff
@@ -20,6 +20,22 @@ def set_user
     @user = User.from_omniauth(auth_hash)
   end
 
+  def keycloak_access_token
+    return nil if auth_hash['credentials'].blank?
+
+    conn = Faraday.new
+    token_url = ENV['KEYCLOAK_TOKEN_URL'] || 'http://localhost:8180/auth/realms/Twilight/protocol/openid-connect/token'
```

Comment:
```suggestion
    token_url = ENV.fetch('KEYCLOAK_TOKEN_URL'])
```

Let's just require the ENV var and set it in .env.development.

Reply:
Okay